sdk/tables/aztable/table_client.go

```diff
@@ -69,70 +70,97 @@ func (t *TableClient) Query(queryOptions *QueryOptions) TableEntityQueryResponse
+	if etag == nil {
+		nilEtag := "*"
+		etag = &nilEtag
+	}
+	return t.client.DeleteEntity(ctx, t.Name, partitionKey, rowKey, *etag, nil, &QueryOptions{})
 }
 
 // UpdateEntity updates the specified table entity if it exists.
 // If updateMode is Replace, the entity will be replaced. This is the only way to remove properties from an existing entity.
 // If updateMode is Merge, the property values present in the specified entity will be merged with the existing entity. Properties not specified in the merge will be unaffected.
 // The specified etag value will be used for optimistic concurrency. If the etag does not match the value of the entity in the table, the operation will fail.
 // The response type will be TableEntityMergeResponse if updateMode is Merge and TableEntityUpdateResponse if updateMode is Replace.
-func (t *TableClient) UpdateEntity(ctx context.Context, entity map[string]interface{}, etag *string, updateMode TableUpdateMode) (interface{}, error) {
-	pk := entity[partitionKey].(string)
-	rk := entity[rowKey].(string)
+func (t *TableClient) UpdateEntity(ctx context.Context, entity []byte, etag *string, updateMode TableUpdateMode) (interface{}, error) {
 	var ifMatch string = "*"
 	if etag != nil {
 		ifMatch = *etag
 	}
+
+	var mapEntity map[string]interface{}
+	err := json.Unmarshal(entity, &mapEntity)
+	if err != nil {
+		return entity, err
+	}
+
+	pk, _ := mapEntity[partitionKey]
+	partKey := pk.(string)
+
+	rk, _ := mapEntity[rowKey]
+	rowkey := rk.(string)
+
 	switch updateMode {
 	case Merge:
-		return t.client.MergeEntity(ctx, t.Name, pk, rk, &TableMergeEntityOptions{IfMatch: &ifMatch, TableEntityProperties: entity}, nil)
+		return t.client.MergeEntity(ctx, t.Name, partKey, rowkey, &TableMergeEntityOptions{IfMatch: &ifMatch, TableEntityProperties: mapEntity}, &QueryOptions{})
 	case Replace:
-		return t.client.UpdateEntity(ctx, t.Name, pk, rk, &TableUpdateEntityOptions{IfMatch: &ifMatch, TableEntityProperties: entity}, nil)
+		return t.client.UpdateEntity(ctx, t.Name, partKey, rowkey, &TableUpdateEntityOptions{IfMatch: &ifMatch, TableEntityProperties: mapEntity}, &QueryOptions{})
 	}
 	return nil, errors.New("Invalid TableUpdateMode")
 }
 
 // UpsertEntity replaces the specified table entity if it exists or creates the entity if it does not exist.
 // If the entity exists and updateMode is Merge, the property values present in the specified entity will be merged with the existing entity rather than replaced.
 // The response type will be TableEntityMergeResponse if updateMode is Merge and TableEntityUpdateResponse if updateMode is Replace.
-func (t *TableClient) UpsertEntity(ctx context.Context, entity map[string]interface{}, updateMode TableUpdateMode) (interface{}, error) {
-	pk := entity[partitionKey].(string)
-	rk := entity[rowKey].(string)
+func (t *TableClient) UpsertEntity(ctx context.Context, entity []byte, updateMode TableUpdateMode) (interface{}, error) {
```

Comment:
does the type change for entity mean that developers always have to marshal their entity before they pass it to us? Could we do it for them?

Reply:
I don't think it right to have asymmetrical behavior where customers MUST unmarshal themself and we marshal for them. I'm not aware of any Go package that does this so the symmetry is prefered.